src/main/java/com/zontik/groshiky/config/security/AppSecurityConfig.java

```diff
@@ -0,0 +1,43 @@
+package com.zontik.groshiky.config.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+@Configuration
+@EnableWebSecurity
+public class AppSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    public AuthenticationProvider authenticationProvider;
+
+    @Bean
+    public PasswordEncoder passwordEncoder(){
+        return new BCryptPasswordEncoder();
+    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth) throws Exception {
+       auth.authenticationProvider(authenticationProvider);
+    }
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http.authorizeRequests()
+                    .antMatchers("/index").permitAll()
+                    .antMatchers("/dashboard").authenticated()
+                .and()
+                    .formLogin()
+                    .loginPage("/index")
+                    .usernameParameter("login")
+                    .loginProcessingUrl("/perform_index")
```

Comment:
Can't figure out how it is supposed to work... I don't see such a method or something that can handle the "/perform_index" URL. Right now it returns 404.